confgenerator/confgenerator_test.go

```diff
@@ -149,6 +149,43 @@ func TestGoldens(t *testing.T) {
 	}
 }
 
+func TestDataprocDefaults(t *testing.T) {
+	t.Parallel()
+
+	goldensDir := "goldens"
+	testName := "builtin"
+	dataprocMetadata := map[string]string{
+		"dataproc-cluster-name": "test-cluster",
+		"dataproc-cluster-uuid": "test-uuid",
+		"dataproc-region": "test-region",
+	}
+
+	t.Run(testName, func(t *testing.T) {
+		t.Parallel()
+		pc := testPlatforms[0]
+		assert.Equal(t, pc.name, "linux")
```

Comment:
This is going to bite someone in the future who is changing the `testPlatforms` array. Just put the first platform into its own global variable and use it in both the array and here.